lib/CodeGen/CGDecl.cpp

```diff
@@ -1092,6 +1092,7 @@ CodeGenFunction::EmitAutoVarAlloca(const VarDecl &D) {
 
     if (!DidCallStackSave) {
       // Save the stack.
+      // XXXAR: Should this really be address space 0?
```

Comment:
I don't know if we have any tests that use the savestack / restorestack intrinsics.  Try putting a VLA in a loop, with the size set to the loop induction variable.  We might get away with the save / restore being an `i8*` - it's currently spilling and loading $sp, though eventually it will spill and restore $c11.